components/custom_status/date_time_input.tsx

```diff
@@ -61,58 +57,87 @@ const Navbar: React.FC<Partial<NavbarElementProps>> = (navbarProps: Partial<Navb
-                    viewBox='0 0 10 16'
-                    fill='none'
-                    xmlns='http://www.w3.org/2000/svg'
-                >
-                    <path
-                        d='M0.41 2.09605L6.314 8.00005L0.41 13.904L1.994 15.506L9.5 8.00005L1.994 0.494048L0.41 2.09605Z'
-                        fill='#3D3C40'
-                        fillOpacity='0.56'
-                    />
-                </svg>
+                <i
+                    className='fa fa-angle-right'
+                    aria-hidden='true'
+                />
             </button>
         </div>
     );
 };
 
-const DateTimeInputContainer: React.FC = () => {
-    const [day, setDay] = useState('Today');
-    const currentUserId = useSelector(getCurrentUserId);
-    const userTimezone = useSelector((state: GlobalState) => getUserTimezone(state, currentUserId));
-    const locale = useSelector(getCurrentLocale);
+const {CUSTOM_STATUS_TIME_PICKER_INTERVALS_IN_MINUTES} = Constants;
+export function getRoundedTime(value: Date) {
+    const roundedTo = CUSTOM_STATUS_TIME_PICKER_INTERVALS_IN_MINUTES;
+    const start = moment(value);
+    const diff = start.minute() % roundedTo;
+    if (diff === 0) {
+        return value;
+    }
+    const remainder = roundedTo - diff;
+    return moment(start).add(remainder, 'm').seconds(0).milliseconds(0).toDate();
+}
+
+const getDateInIntervals = (startTime: Date, interval: number): Date[] => {
+    let time = startTime;
+    const nextDay = moment(startTime).add(1, 'days').startOf('day').toDate();
+    const intervals: Date[] = [];
+    while (time < nextDay) {
+        intervals.push(time);
+        time = moment(time).add(interval, 'minutes').seconds(0).milliseconds(0).toDate();
+    }
+
+    return intervals;
+};
+
+type Props = {
+    time: Date;
+    handleChange: (date: Date) => void;
+    timezone?: string;
+}
 
-    const enableTimezone = useSelector(areTimezonesEnabledAndSupported);
+const DateTimeInputContainer: React.FC<Props> = (props: Props) => {
+    const locale = useSelector(getCurrentLocale);
+    const {time, handleChange, timezone} = props;
+    const [timeOptions, setTimeOptions] = useState<Date[]>([]);
 
-    let currentDate;
-    if (enableTimezone) {
-        if (userTimezone.useAutomaticTimezone) {
-            currentDate = getCurrentDateForTimezone(userTimezone.automaticTimezone);
+    const setTimeAndOptions = () => {
+        const currentTime = timezone ? getCurrentDateAndTimeForTimezone(timezone) : new Date();
+        let startTime: Date;
+        if (time.getDate() === currentTime.getDate()) {
+            startTime = getRoundedTime(currentTime);
         } else {
-            currentDate = getCurrentDateForTimezone(userTimezone.manualTimezone);
+            startTime = moment(time).startOf('day').toDate();
         }
-    } else {
-        currentDate = new Date();
-    }
+        setTimeOptions(getDateInIntervals(startTime, CUSTOM_STATUS_TIME_PICKER_INTERVALS_IN_MINUTES));
```

Comment:
You are passing the value `CUSTOM_STATUS_TIME_PICKER_INTERVALS_IN_MINUTES` as a parameter in `getDateInIntervals`, but using it directly from constants in `getRoundedTime`.

Why is this behavior inconsistent?